torch/csrc/jit/passes/lower_tuples.cpp

```diff
@@ -24,26 +24,95 @@ std::unordered_set<Symbol> supported_ops = {
     prim::Return,
     prim::PythonOp,
     aten::format,
-};
+    prim::Uninitialized,
+    aten::__getitem__};
+
+// Flatten block inputs and insert a tuple construct in the block
+static void flattenTupleInBlockParam(Node* n, size_t index) {
+  auto input = n->inputs()[index];
+  TupleTypePtr tt = input->type()->cast<TupleType>();
+
+  Block* block = n->blocks().at(0);
+  Node* block_node = n;
+
+  std::vector<Value*> new_node_inputs = {};
+  auto new_construct_node =
+      block->prependNode(block->owningGraph()->create(prim::TupleConstruct));
+  for (size_t j = 0; j < tt->elements().size(); ++j) {
+    auto new_block_in = block->addInput();
+    new_construct_node->addInput(new_block_in);
+    block_node->addInput(input->node()->inputs().at(j));
```

Comment:
Do you keep the order of the `block_node` input? The `addInput` here should be inserted in the middle of the `block_node` inputs, right? I thought we need insert input starting from `index`, see below, you `block->insertOutput(index + j + 1, input->node()->inputs().at(j));`. why there is no `insertInput(index + j + 1)` here?

Reply:
Good point, fixed this.